tests/func/test_remote.py

```diff
@@ -232,6 +232,13 @@ def test_modify_missing_remote(tmp_dir, dvc):
     assert main(["remote", "modify", "myremote", "user", "xxx"]) == 251
 
 
+def test_modify_parent_remote(tmp_dir, dvc):
+    assert main(["remote", "add", "myremote", "http://example.com/path"]) == 0
+    assert (
+        main(["remote", "modify", "myremote", "user", "xxx", "--local"]) == 0
+    )
```

Comment:
Should we also try reading the config? I understand that the issue was during `modify` but I don't want to trust that.

Reply:
Good point! Added more throughout checks.